src/backend/api/meals.js

```diff
@@ -1,15 +1,81 @@
 const express = require("express");
+const { max } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
-  try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
-  } catch (error) {
-    throw error;
+  let myTable = knex ("Meal");
+  //Get meals that has a price smaller than maxPrice
+  if ("maxPrice" in request.query) {
+    myTable = myTable.where("price", "<=", request.query.maxPrice);
   }
+  //Get meals that still has available reservations
```

Comment:
This handler is mostly good, it handles combining any number of options 👍🏼 
And I like how you push most of the work into knex 👌🏼 

Bonus:
It's still fairly heavy on the usage of if to control the flow which can be hard to understand since it forces you to jump between "conditional" and "knex" code.
It would be great if we could split those two steps apart.

You can look at [this ](https://github.com/negaren/meal-sharing/pull/1/files#r826793745) comment for hints.